.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,33 @@
+name: github pages
+
+on:
+  push:
+    branches:
+      - main  # Set a branch to deploy
+  pull_request:
```

Comment:
Do we want to run it on every PR?

Reply:
Yeah, I was just wondering if we could just run it on PRs that change the docs folder, as I thought it would not mean much on other PRs. But I was not sure if that's possible, and I think the current setup works fine!